lib/active_admin_importer/import.rb

```diff
@@ -50,17 +56,21 @@ def valid?
 
     private
 
+    def log_error(message)
+      ::Rails.logger.error("[IMPORT: #{@csv_file.md5}]: #{message}")
+    end
+
+    def log_info(message)
+      ::Rails.logger.info("[IMPORT: #{@csv_file.md5}]: #{message}")
+    end
+
     def process_row(row)
       @current_row += 1
       log_info("IMPORTING ROW - #{current_row}")
       data = row.to_hash
 
       if data.present?
-        if @block
-          @block.call(@model, data, @controller)
-        else
-          @model.create!(data)
-        end
+        @definition[:each_row].call(data, self)
```

Comment:
yea both are available now. instead of 3 params it's now 2, the first being the row and the second being the import instance, through which you can access everything, import.controller, import.model